js/foundation.offcanvas.js

```diff
@@ -328,6 +324,12 @@ class OffCanvas extends Plugin {
       this.$overlay.addClass('is-closable');
     }
 
+    /**
+     * Fires when the off-canvas menu opens.
+     * @event OffCanvas#opened
+     */
+    this.$element.trigger('opened.zf.offcanvas');
+
```

Comment:
I think this should be at the very end, after every class/attributes has been set, so after `this._addContentClasses`. Look at the others components for reference.

Reply:
Also after the return statement which closes it early?

Reply:
> Also does IE 11 works with 3e86d6e ? If yes, what was the problem ?

No, I just reverted because it clearly makes not much sense when it is a browser bug ;-)
Not reproducible with IE11 in our own browserstack instance

Reply:
https://codepen.io/DanielRuf/pen/XZeqjO was used for testing

Reply:
> Also after the return statement which closes it early?

Well, I guess yes.

Reply:
> Also after the return statement which closes it early?

The return statement is inside the `transitionend` event. It has no impact on the `opaned.zf.offcanvas` and runs way after it.

Reply:
Right, saw it also. The best and cleanest solution is to trigger the event always after adding all needed classes as triggering events and adding classes (when the event is triggered before this) produces race conditions as far as I can see.